c7n/resources/rds.py

```diff
@@ -1241,3 +1241,125 @@ class resource_type(object):
         filter_type = 'scalar'
         dimension = None
         date = None
+
+
+@filters.register('db-parameter')
+class ParameterGroupFilter(ValueFilter):
+    """
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: rds-parametergroup-example
+                resource: rds
+                filters:
+                  - type: db-parameter
+                    key: someparam
+                    op: eq
+                    value: someval
+
+    based on the `describe_db_instances API`_
+
+    .. _describe_db_instances API: http://boto3.readthedocs.io/en/latest/\
+    reference/services/rds.html#RDS.Client.describe_db_instances
+    """
+
+    schema = {
+        'type': 'object',
+        # Doesn't mix well with inherits that extend
+        'additionalProperties': False,
+        'required': ['type', 'key', 'op', ],  # omit value for future "isnull"
+        'properties': {
+            'type': {'enum': ['db-parameter']},
+            'key': {'type': 'string'},
+            'default': {'oneOf': [
+                {'type': 'string'},
+                {'type': 'boolean'},
+                {'type': 'number'}
+            ]},
+            'value': {'oneOf': [
+                {'type': 'string'},
+                {'type': 'boolean'},
+                {'type': 'number'}
+            ]},
+            'op': {'enum': OPERATORS.keys()}}}
+
+    permissions = ('rds:DescribeDBInstances', 'rds:DescribeDBParameters', )
+
+    def process(self, resources, event=None):
+        pkey = self.data.get('key', '')
+        op = OPERATORS.get(self.data.get('op', 'eq'))
+        target_value = self.data.get('value', '0')
+        default = self.data.get('default', '0')
+        results = []
+
+        client = local_session(self.manager.session_factory).client('rds')
+        for resource in resources:
+            for pg in resource['DBParameterGroups']:
+                group_name = pg[u'DBParameterGroupName']
+
+                # get the params for that group
+                if 'Parameters' in pg:  # pg.has_key('Parameters'):
+                    # use previously augmented data
+                    db_params = pg['Parameters']
+                else:
+                    db_params = client.describe_db_parameters(
```

Comment:
parameter groups are commonly shared between multiple db instances, this should up front calculate and fetch the groups instead of fetching it redundantly per resource.